File: Data Structures & Algorithms/jump-game/submission-4.java
class Solution {
    public boolean canJump(int[] nums) {

        int n =nums.length;
        return dfs(nums,0);
    }

    boolean dfs(int[] nums, int jmp){
        
        if(jmp>=nums.length-1) return true;
        if(nums[jmp]==0) return false;
        int temp=jmp+nums[jmp];
        boolean res = false;
        for(int i=jmp+1;i<=temp && i<nums.length;i++){
            res = res || dfs(nums,i);
        }
        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        f = Counter(nums)
        
        heap = []

        for num,freq in f.items():
            heapq.heappush(heap,(freq,num))
            if len(heap)>k:
                heapq.heappop(heap)

        res =[]
        
        for i in range(k):
            ele = heapq.heappop(heap)
            res.append(ele[1])
    

        return res
